### PR TITLE
Updated frontend design with tailwind support

### PR DESCRIPTION
**Pull request by seoyoungcho213**
## Rationale
Updated frontend design with Tailwind CSS.

## Usage
Downloading Tailwind CSS is necessary. After download, you are good to go with `npm run build-css` command in the terminal.

## Changes
All page designs including main, login, signup, dashboard, and settings.
Updated dependency in README.

### Non-trivial Files
N/A

### Testing

- [ ] ➕ Added units to existing tests
- [ ] 🐣 Started a new group of tests
- [x] 🙅 New tests aren't needed
- [ ] 🙋 Need help to add tests

### Dependencies
**Tailwind CSS**
`npm install -D tailwindcss
npx tailwindcss init
npm run build-css`

If your code doesn't work as expected, please let me know.

### Documentation Changes

- [x] 📜 README.md
- [x] 💬 Added **necessary** comments to existing code
- [ ] 🙅 no documentation needed

## Issues and Bugs
Navigation bar text moves when user signs in.

## Possible Solutions
modify HTML settings

## Additional Notes
Let me know if you find any other bugs with the page !!
Also, here is the image of expected main page: 
image

### CODE REVIEW

File: internal/templates/index.go.html [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-        <button class="btn2" type="submit">Login</button>
-    </a>
-</div>
+{{ template  "navbar" . }}
+<body>
+<section class="px-8 py-24">
+    <div class="container mx-auto text-center">
+        <h1 class="block antialiased tracking-normal font-sans text-4xl font-semibold leading-[1.3] text-accent-700
+        mb-4">
+            Welcome to MATCHA !</h1>
```
Comment (CarlosACJ55) [inline internal/templates/index.go.html:10]: I think either make it all in one line or put the </h1> on the next line

File: internal/templates/index.go.html [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+        <p class="mb-8 block font-sans text-base font-normal leading-relaxed antialiased text-accent-300">
+            Matcha is the first truly comprehensive, web-based personal finance tracker and budgeting tool.</p>
```
Comment (CarlosACJ55) [inline internal/templates/index.go.html:12]: Put </p> on the next line

**Review by CarlosACJ55 — CHANGES_REQUESTED** [review]
consider and decide:

File: .idea/runConfigurations/go_build_github_com_matcha_devs_matcha.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <module name="matcha" />
     <working_directory value="$PROJECT_DIR$" />
     <envs>
-      <env name="MYSQL_PASSWORD" value="" />
+      <env name="MYSQL_PASSWORD" value="xO|I]Br8n}56" />
```
[the remark this answers is not in the document]
Comment (seoyoungcho213) [reply .idea/runConfigurations/go_build_github_com_matcha_devs_matcha.xml:6]: 😢

File: internal/templates/dashboard.go.html [hunk 4]
```diff
@@ -1,20 +1,17 @@
...
-    <button hx-post="/logout" class="btn">Log out</button>
-</div>
-
-{{ . }}
+<section class="px-8 py-24">
+    <div class="container mx-auto text-center">
+        <h1 class="block antialiased tracking-normal font-sans text-4xl font-semibold leading-[1.3] text-accent-300
+        mb-4">
+            Hello, {{ .User.FirstName }} !</h1>
+        <p class="mb-8 block font-sans text-base font-normal leading-relaxed antialiased text-accent-700">
```
[the remark this answers is not in the document]
Comment (seoyoungcho213) [reply internal/templates/dashboard.go.html:11]: It is already reformatted. But I reformatted other files!

File: internal/templates/index.go.html (hunk 1 above)
Comment (seoyoungcho213) [reply internal/templates/index.go.html:10]: Even tho its against the formatting?

**Comment by seoyoungcho213** [comment]
So, I actually also did some research on that. While it is more cleaner and lighter memory wise when you use the executable file from their github, Tailwind CSS officially recommend downloading through npm if we are going to launch a service. Also, our UI is not designed yet, so we might add multiple features that doesn't exist in current version. If this happens multiple times, we would have to set up multiple times to get through this issue.

For that reason, I would prefer option 1.

**Comment by CarlosACJ55** [comment]
Awesome, I'm loving how the PR turned out. Maybe we should add UI screenshots to the end of the README.md?

**Comment by seoyoungcho213** [comment]
Will do that! If there is no feedback other than that, is it okay for me to merge & close this PR?

**Review by CarlosACJ55 — APPROVED** [review]
Perfect

**Review by CarlosACJ55 — APPROVED** [review]

**Pull request merged**